contracts/sdjwt-verifier/src/types.rs

```diff
@@ -79,60 +86,97 @@ pub enum MathsOperator {
                     }
                     (serde_json::Value::Number(p_val), Criterion::Number(c_val, op)) => {
                         if p_val.is_u64() {
                             let num = p_val.as_u64().unwrap();
-
                             match op {
                                 MathsOperator::GreaterThan => {
                                     if num <= c_val {
-                                        return Ok(false);
-                                    };
+                                        return Err(
+                                            SdjwtVerifierResultError::CriterionValueFailed(key),
+                                        );
+                                    }
                                 }
                                 MathsOperator::LessThan => {
                                     if num >= c_val {
-                                        return Ok(false);
-                                    };
+                                        return Err(
+                                            SdjwtVerifierResultError::CriterionValueFailed(key),
+                                        );
+                                    }
                                 }
                                 MathsOperator::EqualTo => {
                                     if num != c_val {
-                                        return Ok(false);
-                                    };
+                                        return Err(
+                                            SdjwtVerifierResultError::CriterionValueFailed(key),
+                                        );
+                                    }
                                 }
                             }
                         } else {
-                            return Err(SdjwtVerifierError::CriterionValueNumberInvalid);
+                            return Err(SdjwtVerifierResultError::CriterionValueNumberInvalid);
                         }
                     }
                     (serde_json::Value::Bool(bool_val), Criterion::Boolean(c_val)) => {
                         if bool_val != &c_val {
-                            return Ok(false);
+                            return Err(SdjwtVerifierResultError::CriterionValueFailed(key));
                         }
                     }
-                    _ => return Err(SdjwtVerifierError::CriterionValueTypeUnexpected),
+                    _ => return Err(SdjwtVerifierResultError::CriterionValueTypeUnexpected),
                 };
             } else {
-                return Err(SdjwtVerifierError::DisclosedClaimNotFound(key));
+                return Err(SdjwtVerifierResultError::DisclosedClaimNotFound(
+                    key.to_string(),
+                ));
             }
         }
+        // For when there are no dislosure required
+        Ok(())
+    } else if let serde_json::Value::Null = verified_claims {
+        if presentation_request.is_empty() {
```

Comment:
Consider Extra conditions here https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html#extra-conditionals-with-match-guards

Reply:
fixed in https://github.com/nymlab/AVIDA/tree/2d03896591111ec65986b15568dd90fb23bf0b3e